common/thorhelper/thorsoapcall.cpp

```diff
@@ -2469,9 +2469,14 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                     checkTimeLimitExceeded(&remainingMS);
                     Url &connUrl = master->proxyUrlArray.empty() ? url : master->proxyUrlArray.item(0);
 
+                    CCycleTimer dnsTimer;
+
                     // TODO: for DNS, do we use timeoutMS or remainingMS or remainingMS / maxRetries+1 or ?
                     ep.set(connUrl.host.get(), connUrl.port, master->timeoutMS);
 
+                    master->logctx.noteStatistic(StTimeSoapcallDNS, dnsTimer.elapsedNs());
```

Comment:
Worth assigning elapsedNs to a temporary and then dividing for the next one to avoid an extra call to get_cycles_now?  
Actually, probably better to record Ns in the span - that is what the general stat processing does.

Reply:
I thought about that so just one call.  I can do that.
But for the Span - its just an attribute so I didn't think any processing was done on it ?
Do you mean use "Ns" label in the Span attribute as well and then use the same value as the stat ?